code/modules/clothing/masks/gasmask.dm

```diff
@@ -2,25 +2,82 @@
 	name = "gas mask"
 	desc = "A face-covering mask that can be connected to an air supply. While good for concealing your identity, it isn't good for blocking gas flow." //More accurate
 	icon_state = "gas_alt"
-	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	flags_inv = HIDEEARS|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
 	w_class = WEIGHT_CLASS_NORMAL
 	inhand_icon_state = "gas_alt"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH | PEPPERPROOF
 	resistance_flags = NONE
+	///Max numbers of installable filters
+	var/max_filters = 1
+	///List to keep track of each filter
+	var/list/gas_filters
+
+/obj/item/clothing/mask/gas/Initialize()
+	. = ..()
+	if(!max_filters)
+		return
+	for(var/i in 1 to max_filters)
+		var/obj/item/gas_filter/filter = new(src)
+		LAZYADD(gas_filters, filter)
+	has_filter = TRUE
+
+/obj/item/clothing/mask/gas/examine(mob/user)
+	. = ..()
+	if(max_filters > 0)
+		. += "<span class='notice'>[src] has [max_filters] slot\s for filters.</span>"
+	if(LAZYLEN(gas_filters) > 0)
+		. += "<span class='notice'>Currently there [LAZYLEN(gas_filters) == 1 ? "is" : "are"] [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
+
+/obj/item/clothing/mask/gas/attackby(obj/item/filter, mob/user)
+	. = ..()
+	if(!istype(filter, /obj/item/gas_filter))
+		return TRUE
+	if(LAZYLEN(gas_filters) >= max_filters)
+		return TRUE
+	if(!user.transferItemToLoc(filter, src))
+		return TRUE
+	LAZYADD(gas_filters, filter)
+	if(LAZYLEN(gas_filters) > 0)
+		has_filter = TRUE
```

Comment:
why the lazylen check?

Reply:
uh i added as a check to ensure that it has filters, but i see how that's dumb so i'll remove it